### PR TITLE
8305942: Open source several AWT Focus related tests

### PR DESCRIPTION
**Pull request by prsadhuk**
This opensources several other AWT Focus related tests.
CI runs are ok.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8305942](https://bugs.openjdk.org/browse/JDK-8305942): Open source several AWT Focus related tests

### Reviewers
 * [Phil Race](https://openjdk.org/census#prr) (@prrace - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/13527/head:pull/13527` \
`$ git checkout pull/13527`

Update a local copy of the PR: \
`$ git checkout pull/13527` \
`$ git pull https://git.openjdk.org/jdk.git pull/13527/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 13527`

View PR using the GUI difftool: \
`$ git pr show -t 13527`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/13527.diff">https://git.openjdk.org/jdk/pull/13527.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/13527#issuecomment-1514414718)

### CODE REVIEW

File: test/jdk/java/awt/Focus/QuickTypeTest.java [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+    static JTextField tf1;
+    static JTextField tf2;
+    static SmartKeyAdapter ska;
+    static Object keyMonitor;
+
+    public static void main(String[] args) throws Exception {
+        try {
+            EventQueue.invokeAndWait(() -> {
+                // Check for Win32 environment. This test is only for Win32.
+                Properties prop = System.getProperties();
```
Comment (prrace) [inline test/jdk/java/awt/Focus/QuickTypeTest.java:67]: Is it really ?

File: test/jdk/java/awt/Focus/TemporaryLostComponentDeadlock.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+            frame1 = new Dialog(frame, "Frame 1", false);
+            frame1.add(new Button("focus owner"));
+            frame1.pack();
+            frame1.setLocationRelativeTo(null);
+            frame1.setVisible(true);
+        });
+
+        Thread t1 = new Thread() {
+            public void run() {
+                synchronized(frame1) {
```
Comment (turbanoff) [inline test/jdk/java/awt/Focus/TemporaryLostComponentDeadlock.java:52]:
```suggestion
                synchronized (frame1) {
```

File: test/jdk/java/awt/Focus/TemporaryLostComponentDeadlock.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+                    frame1.dispose();
+                    synchronized(frame1) {
+                        frame1.notify();
+                    }
+                }
+            }
+        };
+        try {
+            synchronized(frame1) {
```
Comment (turbanoff) [inline test/jdk/java/awt/Focus/TemporaryLostComponentDeadlock.java:61]:
```suggestion
            synchronized (frame1) {
```

File: test/jdk/java/awt/Focus/TemporaryLostComponentDeadlock.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+                t1.start();
+                frame1.wait();
+            }
+        } catch( InterruptedException ie) {
```
Comment (turbanoff) [inline test/jdk/java/awt/Focus/TemporaryLostComponentDeadlock.java:65]:
```suggestion
        } catch (InterruptedException ie) {
```

File: test/jdk/java/awt/Focus/QuickTypeTest.java [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+/*
+  @test
+  @bug 4423838
+  @summary KEY_TYPED and KEY_PRESSED generated by the same key are notified to different fi
```
[the remark this answers is not in the document]
Comment (prsadhuk) [reply test/jdk/java/awt/Focus/QuickTypeTest.java]: ok

File: test/jdk/java/awt/Focus/QuickTypeTest.java (hunk 1 above)
Comment (prsadhuk) [reply test/jdk/java/awt/Focus/QuickTypeTest.java]: Took the test as it is..maybe at the time of test being wriiten, there was some issue with other platforms..
seems to be working on mach5 headful platforms, so remove the windows restriction

**Review by prrace — APPROVED** [review]

**Pull request closed without merging**